src/lib/route-linestring.js

```diff
@@ -23,33 +23,44 @@ export async function loadLookupTables() {
 export function replacePtRouteLinesWithHighres(routes) {
   if (lookupTables == null) return routes;
 
-  const {
-    stopTripShapeLookup,
-    shapeIdLineStringLookup,
-  } = lookupTables;
-
-  for (const path of routes.paths) {
-    for (const leg of path.legs) {
-      if (leg.type === 'pt'){
-        const startStop = leg.stops[0];
-        const endStop = leg.stops[leg.stops.length -1];
-
-        const routeId = leg['route_id'];
-        const tripId = leg['trip_id'];
-        const shapeId = stopTripShapeLookup[routeId][tripId];
-        const shape = shapeIdLineStringLookup[shapeId];
-
-        if (shape) {
-          const fullLineString = lineString(shape);
+  try {
+    const routesCopy = JSON.parse(JSON.stringify(routes));
+    const {
+      stopTripShapeLookup,
+      shapeIdLineStringLookup,
+    } = lookupTables;
   
-          const startPoint = point(startStop.geometry.coordinates);
-          const endPoint = point(endStop.geometry.coordinates);
-          const slicedLine = lineSlice(startPoint, endPoint, fullLineString);
-          leg['geometry'] = slicedLine.geometry;
+    for (const path of routesCopy.paths) {
+      for (const leg of path.legs) {
+        if (leg.type === 'pt'){
+          const startStop = leg.stops[0];
+          const endStop = leg.stops[leg.stops.length -1];
+  
+          const routeId = leg['route_id'];
+          const tripId = leg['trip_id'];
+          const shapeIdsForRoute = stopTripShapeLookup[routeId];
+  
+          if (shapeIdsForRoute) {
```

Comment:
Isn't this the only thing that could be undefined, therefore doesn't this check make the catch-all around everything unnecessary?

Reply:
`lineSlice` can throw if any of the input is malformed in any way. A linestring thats a single point for example.
Since this part of the code deals with data parsed from the gtfs files, which is external and can get botched, this is a good safety net. A much more elegant solution would be doing schema validation on the ingested dictionaries, but eh, thats way too much work for some data that nneds to only be used in one function.

Reply:
Can you put the try/catch around just the `lineSlice` call then?

Reply:
sure, I wrapped all the turf punctions though coz all of them have some validation